sdk/core/Azure.Core.Experimental/src/GeoJson/GeoArray.cs

```diff
@@ -0,0 +1,113 @@
+{
+    /// <summary>
+    /// Represents a geometry coordinates array
+    /// </summary>
+    /// <typeparam name="T">The type of the value.</typeparam>
+    public readonly struct GeoArray<T>: IEnumerable<T>
+    {
+        private readonly object _container;
+
+        internal GeoArray(object container)
+        {
+            _container = container;
+        }
+
+        /// <summary>
+        /// Returns a value at the provided index.
+        /// </summary>
+        /// <param name="index">The index to retrieve the value from.</param>
+        public T this[int index]
+        {
+            get
+            {
+                return _container switch
+                {
+                    T[] array => array[index],
+                    GeoPointCollection pointCollection => (T) (object) pointCollection.Points[index].Coordinates,
+                    GeoLineStringCollection lineCollection => (T) (object) lineCollection.Lines[index].Coordinates,
+                    GeoPolygon polygon => (T) (object) polygon.Rings[index].Coordinates,
+                    GeoPolygonCollection polygonCollection => (T) (object) polygonCollection.Polygons[index].Coordinates,
+                    _ => default!
+                };
+            }
+        }
+
+        /// <summary>
+        /// Returns the size of the array.
+        /// </summary>
+        public int Count
+        {
+            get
+            {
+                return _container switch
+                {
+                    T[] array => array.Length,
+                    GeoPointCollection pointCollection => pointCollection.Points.Count,
+                    GeoLineStringCollection lineCollection => lineCollection.Lines.Count,
+                    GeoPolygon polygon => polygon.Rings.Count,
+                    GeoPolygonCollection polygonCollection => polygonCollection.Polygons.Count,
+                    _ => 0
+                };
+            }
+        }
+
+        /// <summary>
+        /// Returns an enumerator that iterates through the collection.
+        /// </summary>
+        /// <returns>An enumerator that can be used to iterate through the collection.</returns>
+        public Enumerator GetEnumerator()
+        {
+            return new Enumerator(this);
+        }
+
+        IEnumerator IEnumerable.GetEnumerator() => GetEnumerator();
+
+        IEnumerator<T> IEnumerable<T>.GetEnumerator() => GetEnumerator();
+
+        /// <summary>
+        /// Enumerates the elements of a <see cref="GeoArray{T}"/>
+        /// </summary>
+#pragma warning disable CA1034
+        public struct Enumerator: IEnumerator<T>
+        {
+            private readonly GeoArray<T> _array;
+            private int _index;
+
+            internal Enumerator(GeoArray<T> array) : this()
+            {
+                _array = array;
+                _index = -1;
+            }
+
+            /// <inheritdoc />
+            public bool MoveNext()
+            {
+                _index++;
```

Comment:
Is it worth tracking version/generation of the parent container, i.e. that it doesn't change. Or are they readonly in nature such that it doesn't matter?

Reply:
> Or are they readonly in nature such that it doesn't matter

^ This.